src/main/java/com/walmartlabs/x12/X12Parser.java

```diff
@@ -34,20 +36,48 @@
     T parse(String sourceData);
 
     /**
-     * parses the source data into a list of segments each line in the source data is a segment
+     * parses the source data into a list of segments 
+     * 1) assume each segment is on separate line
+     * 2) otherwise try default segment delimiter (~)
+     * 3) otherwise try last character (TODO)
      */
     default List<X12Segment> splitSourceDataIntoSegments(String sourceData) {
+        // assume that the source data has 
+        // each segment on a separate line
+        // and that ALL valid EDI / X12 documents
+        // are > 1 segment 
+        List<X12Segment> segments = splitSourceDataIntoSegments(sourceData, "\\r?\\n");
+        if (segments.size() > 1) {
+            return segments;
+        } else {
+            // if there is only one line in the source data
+            // we should attempt to use the segment separator passed in
+            // and see if we can split up this source data
+            segments = splitSourceDataIntoSegments(sourceData, "\\" + DEFAULT_SEGMENT_SEPARATOR);
+            return segments;
+        }
+    }
+    
+    /**
+     * parses the source data into a list of segments 
+     * using the the segment delimiter that was passed in
+     * @param sourceData
+     * @param segmentSeparator a regex to split segments
+     */
+    default List<X12Segment> splitSourceDataIntoSegments(String sourceData, String segmentSeparator) {
         if (StringUtils.isEmpty(sourceData)) {
             return Arrays.asList();
         } else {
-            return Arrays.asList(sourceData.split("\\r?\\n")).stream()
+            String segmentSeparatorRegEx = segmentSeparator;
+            return Arrays.asList(sourceData.split(segmentSeparatorRegEx)).stream()
```

Comment:
this can also be written as
```
 Arrays.stream(sourceData.split(segmentSeparatorRegEx))
            .map(segment -> new X12Segment(segment))
            .collect(Collectors.toList());
```